### PR TITLE
EVG-15097: implement pod REST model

### PR DESCRIPTION
**Pull request by Kimchelly**
JIra: https://jira.mongodb.org/browse/EVG-15097

### Description 
* Add REST API pod model.
* Add data connector method to find pod by ID.
* Include the status when creating a new DB pod from the REST route to create new pods. 
* Slightly modify some of the HTTP status codes returned when checking a pod ID/secret.
* Revendor cocoa.

### Testing 
Normal REST model and data connector tests.

### CODE REVIEW

File: rest/data/pod.go [hunk 1]
```diff
@@ -28,42 +28,77 @@ func (c *DBPodConnector) CreatePod(p restModel.APICreatePod) (*restModel.APICrea
...
 	}
 	if p == nil {
-		return errors.New("pod does not exist")
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    "pod does not exist",
+		}
 	}
 	if secret != p.Secret {
 		return errors.New("incorrect pod secret")
```
Comment (ybrill) [inline rest/data/pod.go:52]: maybe we should return a 403?

File: rest/data/pod_test.go [hunk 2]
```diff
@@ -53,6 +57,26 @@ func (s *podConnectorSuite) TestCreatePod() {
 	s.Require().NoError(err)
 	s.Assert().Equal("secret", podDB.Secret)
 	s.Assert().Equal("env_value", podDB.TaskContainerCreationOpts.EnvVars["env_name"])
+	s.NoError(db.Clear(pod.Collection))
```
Comment (ybrill) [inline rest/data/pod_test.go:60]: isn't this already accomplished by `TeardownTest()`?

File: rest/model/pod.go [hunk 3]
```diff
@@ -70,3 +72,164 @@ func (p *APICreatePod) ToService() (interface{}, error) {
...
+	o.Platform = &opts.Platform
+	o.EnvVars = opts.EnvVars
+	o.EnvSecrets = opts.EnvSecrets
+}
+
+// ToService converts REST API task container creation options into
+// service-layer task container creation options.
+func (i *APIPodTaskContainerCreationOptions) ToService() (*pod.TaskContainerCreationOptions, error) {
+	if i.Platform == nil {
+		return nil, errors.New("missing platform")
```
Comment (ybrill) [inline rest/model/pod.go:196]: maybe instead this case should set platform to ""?

File: rest/route/middleware.go [hunk 4]
```diff
@@ -330,28 +330,19 @@ func (m *podAuthMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, n
 	id := gimlet.GetVars(r)["pod_id"]
 	if id == "" {
 		if id = r.Header.Get(evergreen.PodHeader); id == "" {
-			gimlet.WriteResponse(rw, gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
-				StatusCode: http.StatusUnauthorized,
-				Message:    "missing pod ID",
-			}))
+			gimlet.WriteResponse(rw, gimlet.MakeJSONErrorResponder(errors.New("missing pod ID")))
```
Comment (ybrill) [inline rest/route/middleware.go:333]: maybe these should return a 400?

File: rest/data/pod.go (hunk 1 above)
Comment (Kimchelly) [reply rest/data/pod.go:52]: I fixed it. It's supposed to return 401.

File: rest/data/pod_test.go (hunk 2 above)
Comment (Kimchelly) [reply rest/data/pod_test.go:60]: Fixed. I think this was left over from when I rebased to fix merge conflicts.

File: rest/model/pod.go (hunk 3 above)
Comment (Kimchelly) [reply rest/model/pod.go:196]: The empty string is still an invalid platform. I assume the calling code won't continue if this returns an error.

File: rest/route/middleware.go (hunk 4 above)
Comment (Kimchelly) [reply rest/route/middleware.go:333]: `MakeJSONErrorResponder` returns 400 bad request by default unless you give it a `gimlet.ErrorResponder` as an error.

**Review by ybrill — APPROVED** [review]

**Pull request merged**